lib/fs/basicfs_windows.go

```diff
@@ -150,3 +151,31 @@ func (f *BasicFilesystem) Roots() ([]string, error) {
 
 	return drives, nil
 }
+
+func (f *BasicFilesystem) resolveWin83(absPath string) string {
+	if !strings.Contains(absPath, "~") {
+		return absPath
+	}
+	if in, err := syscall.UTF16FromString(absPath); err != nil {
+		out := make([]uint16, 2*len(absPath))
```

Comment:
Worth allocating more space? Due to the test on L156, we know that we've got an 8.3 path and we're going to end up with a long path. Since `out` is a UTF-16 buffer, it has 2 bytes per character. Allocating `2*len(absPath)` means that `out` is the same length as `in`, i.e. you'll have to reallocate if `out` is even 1 char longer than `in`. Since we know that `out` will definitely be longer than `in`, this means we'll always reallocate. Perhaps this should be `2*len(in)`?